helix-rest/src/test/java/org/apache/helix/rest/server/TestZooKeeperAccessor.java

```diff
@@ -192,4 +191,33 @@ public void testGetStat() throws IOException {
     // Clean up
     _testBaseDataAccessor.remove(path, AccessOption.PERSISTENT);
   }
+
+  @Test
+  public void testDelete() throws IOException {
+    String path = "/path";
+    String deletePath = path + "/delete";
+
+    try {
+      // 1. Create a persistent node. Delete shall fail.
+      _testBaseDataAccessor.create(deletePath, null, AccessOption.PERSISTENT);
+      // Verify with the REST endpoint
+      new JerseyUriRequestBuilder("zookeeper{}?command=delete").format(deletePath)
+          .expectedReturnStatusCode(Response.Status.FORBIDDEN.getStatusCode());
+      Assert.assertTrue(_testBaseDataAccessor.exists(deletePath, AccessOption.PERSISTENT));
+
+      // 2. Create a ephemeral node. Delete shall be done successfully.
+      _testBaseDataAccessor.remove(deletePath, AccessOption.PERSISTENT);
+      _testBaseDataAccessor.create(deletePath, null, AccessOption.EPHEMERAL);
+      // Verify with the REST endpoint
+      String data = new JerseyUriRequestBuilder("zookeeper{}?command=delete").format(deletePath)
+          .isBodyReturnExpected(true).get(this);
+      Map<String, String> result = OBJECT_MAPPER.readValue(data, HashMap.class);
+      Assert.assertEquals(result.get("path"), deletePath);
+      Assert.assertEquals(result.get("delete"), new Boolean(true).toString());
+      Assert.assertFalse(_testBaseDataAccessor.exists(deletePath, AccessOption.PERSISTENT));
```

Comment:
Question here (may not related to your change).
Is the second param here in `exists(String path, int options)` useful? I did not find it being considered in any implementation of BaseDataAccessor.

Reply:
No use at all. There is no such thing in ZK lib. I think we shall clean them up in a separate PR.

Reply:
Right. That confuses me as well. It seems many functions has a not-used 'int option' as input. May I suggest using a value consist with the node type as input? For example, we want to check if the newly created EPHEMERAL node is deleted, let use EPHEMERAL here to avoid further confusion.